### PR TITLE
Remove memory leaks in projection matching when allocating threads data

### PR DESCRIPTION
**Pull request by juannavascalvente**
The thread were allocating memory into a for loop without deallocating the memory. Now it is allocated just once and deallocated when the threads exit.

### CODE REVIEW

File: software/em/xmipp/libraries/reconstruction/angular_projection_matching.h [hunk 1]
```diff
@@ -134,14 +134,12 @@ class ProgAngularProjectionMatching : public XmippProgram
     /** Threads */
     int threads;
     //Numbre of possible orientations
```
Comment (delarosatrevin) [inline software/em/xmipp/libraries/reconstruction/angular_projection_matching.h]: There is a typo here

**Comment by delarosatrevin** [comment]
@juannavascalvente , this is a serious memory leak. Can you check if it is also in v1.0 branch?
The threads allocated memory is freed?
@rmarabini, can you take a look here?

**Comment by juannavascalvente** [comment]
This bug is also in v1.0.

**Comment by delarosatrevin** [comment]
@juannavascalvente , please fix it in v1.0, check with @rmarabini that everything is correct...and we can create a new PR into v1.0. All the bug fixed there are merged later into devel.

**Comment by rmarabini** [comment]
If I am not wrong numOrientations is a parameter introduced by @jvargas.

435b1cfe (jvargas                       2015-09-09 10:03:20 +0200  115)     addParamsLine("  [--number_orientations <numOrientations=1>]  : Number of possible orientations for each experimental image");

Question for @jvargar. Are you interested in this option or should we restore the previous projection matching version?

**Comment by cossorzano** [comment]
I am also using this option in Highres. I would like to keep it

**Pull request merged**